src/OrbitQt/HistogramWidget.cpp

```diff
@@ -146,33 +152,65 @@ static void DrawVerticalAxis(QPainter& painter, const QPoint& axes_intersection,
   return static_cast<double>(histogram.counts[i]) / static_cast<double>(histogram.data_set_size);
 }
 
+static void DrawVerticalHoverLabel(QPainter& painter, const QPoint& axes_intersection,
+                                   const orbit_qt::BarData& bar_data) {
+  QString label_text = QString::fromStdString(absl::StrFormat("%.1f", bar_data.frequency * 100));
+
+  QRect label_rect(QPoint(0, 0), QPoint(kVerticalLabelWidth, kVerticalLabelHeight));
+
+  label_rect.moveTo(axes_intersection.x() - label_rect.width() - kLineWidth / 2 -
+                        kVerticalAxisTickLength - kTickLabelGap,
+                    bar_data.top - label_rect.height() / 2);
+
+  painter.fillRect(label_rect, kHoverLabelColor);
+  painter.drawText(label_rect, Qt::AlignCenter, label_text);
+}
+
 static void DrawHistogram(QPainter& painter, const QPoint& axes_intersection,
                           const orbit_statistics::Histogram& histogram, int horizontal_axis_length,
-                          int vertical_axis_length, double max_freq, uint64_t min_value) {
+                          int vertical_axis_length, double max_freq, uint64_t min_value,
+                          const std::optional<int>& histogram_hover_x) {
   std::vector<int> widths =
       orbit_qt::GenerateHistogramBinWidths(histogram.counts.size(), horizontal_axis_length);
 
-  int left_x = axes_intersection.x() + kLineWidth / 2 +
-               ValueToAxisLocation(histogram.min, horizontal_axis_length, min_value, histogram.max);
   int color_index = 0;
+  std::optional<orbit_qt::BarData> hovered_bar_data;
+  int left = axes_intersection.x() + kLineWidth / 2 +
```

Comment:
Why renaming from left_x to left?

Reply:
Because `_x` is redundant. It's clear left/right correspond to x-axis and top/bottom correspond to y-axis.

Reply:
I would rather prefer to keep them. "Clear" is a strong word.